sdks/python/apache_beam/ml/inference/base_test.py

```diff
@@ -339,6 +375,79 @@ def validate_inference_args(
     third_party_model_handler.batch_elements_kwargs()
     third_party_model_handler.validate_inference_args({})
 
+  def test_run_inference_prediction_result_with_model_id(self):
+    examples = [1, 5, 3, 10]
+    expected = [
+        base.PredictionResult(
+            example=example,
+            inference=example + 1,
+            model_id='fake_model_id_default') for example in examples
+    ]
+    with TestPipeline() as pipeline:
+      pcoll = pipeline | 'start' >> beam.Create(examples)
+      actual = pcoll | base.RunInference(
+          FakeModelHandlerReturnsPredictionResult())
+      assert_that(actual, equal_to(expected), label='assert:inferences')
+
+  @pytest.mark.it_postcommit
+  def test_run_inference_prediction_result_with_side_input(self):
+    test_pipeline = TestPipeline(is_integration_test=True)
+
+    first_ts = math.floor(time.time()) - 30
+    interval = 5
+    main_input_windowing_interval = 7
+
+    # aligning timestamp to get persistent results
+    first_ts = first_ts - (
+        first_ts % (interval * main_input_windowing_interval))
+    last_ts = first_ts + 45
+
+    sample_main_input_elements = ([first_ts - 2, # no output due to no SI
```

Comment:
`no output due to no SI` - This does still get an output, though, right? Since the side input still fires within this window